nixos/modules/i18n/inputMethod/default.nix

```diff
@@ -3,8 +3,20 @@
 with lib;
 let
   cfg = config.i18n.inputMethod;
-  gtk2_file = "/var/gtk2.0/immodules.cache";
-  gtk3_file = "/var/gtk3.0/immodules.cache";
+  gtk2_cache = pkgs.stdenv.mkDerivation {
+    name = "gtk2-immodule.cache";
+    buildInputs = [ pkgs.gtk cfg.package ];
+    buildCommand = ''
+      GTK_PATH=${cfg.package}/lib/gtk-2.0/ gtk-query-immodules-2.0 > $out
+    '';
+  };
+  gtk3_cache = pkgs.stdenv.mkDerivation {
+    name = "gtk3-immodule.cache";
+    buildInputs = [ pkgs.gtk3 cfg.package ];
+    buildCommand = ''
+      GTK_PATH=${cfg.package}/lib/gtk-3.0/ gtk-query-immodules-3.0 > $out
+    '';
+  };
```

Comment:
Add `preferLocalBuild = true; allowSubstitutes = false;` to those -- they look simple enough.